src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -379,6 +390,237 @@ void CascadiaSettings::_LoadDynamicProfiles()
     }
 }
 
+// Method Description:
+// - Searches the local app data folder, global app data folder and app
+//   extensions for json stubs we should use to create new profiles,
+//   modify existing profiles or add new color schemes
+// - If the user settings has any namespaces in the "disabledProfileSources"
+//   property, we'll ensure that the corresponding folders do not get searched
+void CascadiaSettings::_LoadFragmentExtensions()
+{
+    // First, accumulate the namespaces the user wants to ignore
+    std::unordered_set<std::wstring> ignoredNamespaces;
+    const auto disabledProfileSources = CascadiaSettings::_GetDisabledProfileSourcesJsonObject(_userSettings);
+    if (disabledProfileSources.isArray())
+    {
+        for (const auto& json : disabledProfileSources)
+        {
+            ignoredNamespaces.emplace(JsonUtils::GetValue<std::wstring>(json));
+        }
+    }
+
+    // Search through the local app data folder
+    PWSTR localAppDataFolder;
+    SHGetKnownFolderPath(FOLDERID_LocalAppData, 0, nullptr, &localAppDataFolder);
+    auto localAppDataFragments = std::wstring(localAppDataFolder) + FragmentsPath.data();
+    if (std::filesystem::exists(localAppDataFragments))
+    {
+        _ApplyJsonStubsHelper(localAppDataFragments, ignoredNamespaces);
+    }
+
+    // Search through the program data folder
+    PWSTR programDataFolder;
+    SHGetKnownFolderPath(FOLDERID_ProgramData, 0, nullptr, &programDataFolder);
+    auto programDataFragments = std::wstring(programDataFolder) + FragmentsPath.data();
+    if (std::filesystem::exists(programDataFragments))
+    {
+        _ApplyJsonStubsHelper(programDataFragments, ignoredNamespaces);
+    }
+
+    // Search through app extensions
+    // Gets the catalog of extensions with the name "com.microsoft.windows.terminal.settings"
+    const auto catalog = Windows::ApplicationModel::AppExtensions::AppExtensionCatalog::Open(winrt::to_hstring(AppExtensionHostName));
+
+    // Extracting the list of extensions from the catalog is an async operation, but since
+    // we are on a UI thread - and so cannot call a blocking operation like .get() - we use
+    // a mutex and condition variable
+    std::condition_variable cv;
+    std::mutex mtx;

```

Comment:
Would it be easier to just wrap this whole body up in a `fire_and_forget` lambda that hops to the BG thread once at the start, and wait for the cv once at the bottom, rather than hop back and forth between threads a few times?

Reply:
So unfortunately, this part actually needs to be synchronous and so it cannot be in a `fire_and_forget`. We need to wait on the result, but winrt doesn't let us wait on the main thread hence why we manually created our own 'wait' here